address.js

```diff
@@ -125,24 +124,33 @@ proto.extract = function(fieldName, regexes) {
   }
 
   // iterate over the unit regexes and test them against the various parts
-  for (rgxIdx = 0; rgxIdx < regexes.length; rgxIdx++) {
-    for (ii = this.parts.length; ii >= 0; ii-- ) {
+  for (ii = this.parts.length-1; ii >= 0; ii--) {
+    for (rgxIdx = 0; rgxIdx < regexes.length; rgxIdx++) {
+
+      // skip fields which have already been parsed
+      if (this[fieldName]){ continue; }
```

Comment:
Definitely agree that this should be done.  Having a look at the code on `master` I think you can probably return even earlier with this test (i.e. immediately after the variable declarations).  Correct me if I'm wrong though.

Reply:
I'm not really familiar with how all the code fits together.

I can add another isset check above the loops is you think that'd be beneficial.

We'd surely need to leave this one in place too?

Reply:
> We'd surely need to leave this one in place too?

I don't think so.  Adding the following early return close to the top of the function should eliminate any need for doing any additional checking:

```js
// skip fields which have already been parsed
if (this[fieldName]) {
  return this;
}
```

Reply:
Maybe I'm not understanding correctly, but if `fieldName === state` then it's possible to parse state multiple times without the inner isset?

Reply:
I tried removing it and the test I added failed, so I'm going to leave it in for now :)

Reply:
resolved via https://github.com/DamonOehlman/addressit/pull/25/commits/0d286993cd84200058bed0031e434cd3d8a2ee3f

Reply:
Ah yes (it's been a long while since I've worked on the codebase).  You are absolutely correct.  Thank goodness for tests!